Source/objects.cpp

```diff
@@ -670,7 +670,7 @@ void AddObjTraps()
 
 			if (GenerateRnd(2) == 0) {
 				int xp = i - 1;
-				while (!nSolidTable[dPiece[xp][j]]) // BUGFIX: check if xp >= 0
+				while (IsTileNotSolid({ xp, j })) // BUGFIX: check if xp >= 0

```

Comment:
Given this use of the function (and the comment 😀) making `IsTileNotSolid` return false for out of bounds positions would be good.

Reply:
Almost did it when I noticed this case as well, but instead i think it serves as a good reminder to do this later.